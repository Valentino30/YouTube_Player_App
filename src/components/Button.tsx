type ButtonProps = {
  name: string;
  disabled: boolean;
  onClick: () => void;
};

export default function Button({ name, onClick, disabled }: ButtonProps) {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      className="text-white font-medium rounded-lg shadow-md disabled:opacity-50"
    >
      {name}
    </button>
  );
}
