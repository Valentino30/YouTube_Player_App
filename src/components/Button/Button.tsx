type ButtonProps = {
  name: string;
  disabled?: boolean;
  onClick: () => void;
};

export default function Button({ name, onClick, disabled }: ButtonProps) {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      className="text-white font-medium rounded-lg shadow-md disabled:opacity-50 max-w-[200px] truncate"
    >
      {name}
    </button>
  );
}
