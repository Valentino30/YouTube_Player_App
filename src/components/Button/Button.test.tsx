import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Button Component", () => {
  it("renders without errors", () => {
    render(<Button name="Button" onClick={() => {}} />);
    expect(screen.getByText("Button")).toBeInTheDocument();
  });
  it("calls the onClick function when clicked", () => {
    const onClick = vi.fn();
    render(<Button name="Button" onClick={onClick} />);
    screen.getByText("Button").click();
    expect(onClick).toHaveBeenCalled();
  });
  it("is disabled when passed the disabled prop", () => {
    render(<Button name="Button" onClick={() => {}} disabled />);
    expect(screen.getByText("Button")).toBeDisabled();
  });
  it("matches the snapshot", () => {
    const { asFragment } = render(<Button name="Button" onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
