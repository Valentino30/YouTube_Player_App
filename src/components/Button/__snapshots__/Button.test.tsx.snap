// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="text-white font-medium rounded-lg shadow-md disabled:opacity-50 max-w-[200px] truncate"
  >
    Button
  </button>
</DocumentFragment>
`;
